src/app/api/comment/comment.service.ts

```diff
@@ -0,0 +1,44 @@
+import { CreateComment, UpdateComment } from '@/types/dto/comment.dto'
+import { BaseService } from '../core/services/base.service'
+import { PoliciesService } from '../core/services/policies.service'
+import { UserAction } from '../core/types/user'
+import { Resource } from '../core/types/api'
+import { ActivityLog } from '@prisma/client'
+
+export class CommentService extends BaseService {
+  async createComment(data: CreateComment) {
+    const policyGate = new PoliciesService(this.user)
+    policyGate.authorize(UserAction.Create, Resource.Comment)
+
+    const comment = await this.db.comment.create({
+      data: {
+        ...data,
+        taskId: data.taskId,
+        workspaceId: this.user.workspaceId,
+      },
+    })
+
+    return comment
+  }
+
+  async deleteComment(id: string) {
+    const policyGate = new PoliciesService(this.user)
+    policyGate.authorize(UserAction.Delete, Resource.Comment)
+
+    return await this.db.comment.delete({ where: { id } })
+  }
+
+  async updateComment(id: string, data: UpdateComment) {
+    const policyGate = new PoliciesService(this.user)
+    policyGate.authorize(UserAction.Update, Resource.Comment)
+
+    const updatedComment = await this.db.comment.update({
+      where: { id },
+      data: {
+        ...data,
+      },
+    })
+
+    return updatedComment
+  }
+}
```

Comment:
Do we also need a controller to fetch all comments for a particular task, or will that be done through the activity log controller?

Reply:
I have added a GET endpoint for combined response in the activity controller. This e19bafa0201a73e250facc984c7f1e47cbf50771 commit separates the service logic.